chunky/src/java/se/llbit/chunky/renderer/renderdump/FloatingPointCompressor.java

```diff
@@ -149,52 +151,47 @@ public double decodeSingle(byte header, BufferedInputStream input) throws IOExce
   }
 
   public static void compress(
+    SampleBuffer input,
     OutputStream output,
-    double[] input,
-    IntConsumer pixelProgress
+    LongConsumer pixelProgress
   ) throws IOException {
     try (BufferedOutputStream out = new BufferedOutputStream(output)) {
-      if (input.length % 3 != 0)
-        throw new IllegalArgumentException("Dump doesn't have a multiple of 3 values");
-
-      int pixels = input.length / 3;
-      int size = pixels - 1;
-
+      long size = input.numberOfPixels() - 1;
```

Comment:
I like this change! Could you do the same in the `decompress` method?

Reply:
Will do :+1: